core/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java

```diff
@@ -200,6 +207,9 @@ public QueryStringQueryBuilder(StreamInput in) throws IOException {
         timeZone = in.readOptionalTimeZone();
         escape = in.readBoolean();
         maxDeterminizedStates = in.readVInt();
+        if (in.getVersion().onOrAfter(V_5_1_0_UNRELEASED)) {
+            splitOnWhitespace = in.readBoolean();
+        }
```

Comment:
I'd probably add an `else` clause that sets `splitOnWhitespace` to the appropriate value just to be super clear.